src/plopp/backends/matplotlib/static.py

```diff
@@ -15,29 +15,44 @@ def _make_svg_repr(fig):
 REPR_MAP = {'png': _make_png_repr, 'svg': _make_svg_repr}
 
 
+def get_repr_maker(form=None, npoints=0):
+    """
+    Get the function used to create the mimebundle representation of a figure,
+    based on the format requested and the number of points in the figure.
+    """
+    if form is None:
+        form = 'png' if (npoints > 20_000) else 'svg'
+    return REPR_MAP[form.lower()]
+
+
 class StaticFig:
     """
     Create a static figure to represent one-dimensional data.
     """
 
     def __init__(self, *args, FigConstructor, **kwargs):
```

Comment:
And just out of curiosity, is there a specific reason that `FigConstructor` is placed between `*args` and `**kwargs`...?
If any of `kwargs` for `FigConstructor` has the key called `FigConstructor`, it will not be passed, right...?
Or was this the intentino not to have unexpected nested calls...?

Reply:
Not sure how else you would do it?
`FigConstructor` needs to be a mandatory argument, so I thought this is the place to put it?

Reply:
It could be the first positional argument, I thought.
Because, when you are passing all keywords, this could happen.
```
class ClumpsyFig:
    def __init__(self, *, FigConstructor: bool):
        self._figure_constructor = FigConstructor

class Figure:
    def __init__(self, *args, FigConstructor, **kwargs):
        self._figure = FigConstructor(*args, **kwargs)

wrapped_fig = Figure(FigConstructor=ClumpsyFig) 
# TypeError: __init__() missing 1 required keyword-only argument: 'FigConstructor'
```
But... there is very low chance that `matplotlib` or other plotting libraries
to add an argument with the name `FigConstructor`, so maybe it was too much of worry ...?

Another good thing about having the `FigConstructor` as the first positional argument might be
that it will have a similar syntax as `functools.partial`
, which means it passes `*args` and `**kwargs`` to the wrapped callable object.

Reply:
It would however mean that when I use it, I have to do
`return backends.figure1d(FigLine, *args, **kwargs)`
instead of
`return backends.figure1d(*args, FigConstructor=FigLine, **kwargs)`
which may be slightly less readable?

Reply:
Or maybe it's more readable, I don't know anymore...

Reply:
Sorry I think I should have asked in a separate issue : D....